app/src/main/java/protect/card_locker/MainActivity.kt

```diff
@@ -1,585 +1,607 @@
-            public void onTabUnselected(TabLayout.Tab tab) {
+        binding = MainActivityBinding.inflate(getLayoutInflater())
+        setContentView(binding!!.getRoot())
+        Utils.applyWindowInsets(binding!!.getRoot())
+        setSupportActionBar(binding!!.toolbar)
+        groupsTabLayout = binding!!.groups
+        contentMainBinding = ContentMainBinding.bind(binding!!.include.getRoot())
 
-            }
+        mDatabase = DBHelper(this).getWritableDatabase()
 
-            @Override
-            public void onTabReselected(TabLayout.Tab tab) {
+        mUpdateLoyaltyCardListRunnable = Runnable {
+            updateLoyaltyCardList(false)
+        }
 
+        groupsTabLayout!!.addOnTabSelectedListener(object : OnTabSelectedListener {
+            override fun onTabSelected(tab: TabLayout.Tab) {
+                selectedTab = tab.getPosition()
+                Log.d("onTabSelected", "Tab Position " + tab.getPosition())
+                mGroup = tab.getTag()
+                updateLoyaltyCardList(false)
+                // Store active tab in Shared Preference to restore next app launch
+                val activeTabPref = getApplicationContext().getSharedPreferences(
+                    getString(R.string.sharedpreference_active_tab),
+                    MODE_PRIVATE
+                )
+                val activeTabPrefEditor = activeTabPref.edit()
+                activeTabPrefEditor.putInt(
+                    getString(R.string.sharedpreference_active_tab),
+                    tab.getPosition()
+                )
+                activeTabPrefEditor.apply()
             }
-        });
 
-        mHelpSection = contentMainBinding.helpSection;
-        mNoMatchingCardsText = contentMainBinding.noMatchingCardsText;
-        mNoGroupCardsText = contentMainBinding.noGroupCardsText;
-        mCardList = contentMainBinding.list;
-
-        mAdapter = new LoyaltyCardCursorAdapter(this, null, this, mUpdateLoyaltyCardListRunnable);
-        mCardList.setAdapter(mAdapter);
-        registerForContextMenu(mCardList);
-
-        mBarcodeScannerLauncher = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
-            // Exit early if the user cancelled the scan (pressed back/home)
-            if (result.getResultCode() != RESULT_OK) {
-                return;
+            override fun onTabUnselected(tab: TabLayout.Tab?) {
             }
 
-            Intent editIntent = new Intent(getApplicationContext(), LoyaltyCardEditActivity.class);
-            editIntent.putExtras(result.getData().getExtras());
-            startActivity(editIntent);
-        });
+            override fun onTabReselected(tab: TabLayout.Tab?) {
+            }
+        })
+
+        mHelpSection = contentMainBinding!!.helpSection
+        mNoMatchingCardsText = contentMainBinding!!.noMatchingCardsText
+        mNoGroupCardsText = contentMainBinding!!.noGroupCardsText
+        mCardList = contentMainBinding!!.list
```

Comment:
I think we can just get rid of these variables, Kotlin is a lot smarter about types and directly calling the elements in the `contentMainBinding` will probably be fine in Kotlin and even more readable and save some code complexity.